addon/lib/toolbar-button.js

```diff
@@ -23,11 +26,22 @@ let settings;
 let button;
 let panel;
 let collapsed;
+let prefs;
 
-function setActionButton(dark) {
-  const iconPrefix = dark ? './icon-inverted' : './icon';
+function themeChanged() {
+  setTimeout(setButton);
+}
+
+function isLightTheme() {
+  var chromeWindow = viewFor(windows.activeWindow);
+  return !chromeWindow.document.getElementById('nav-bar').matches('[brighttext]');
+}
 
-  button = ActionButton({ // eslint-disable-line new-cap
+function setButton() {
+  const iconPrefix = isLightTheme() ? './icon' : './icon-inverted';
+  if (button) { button.destroy(); }
```

Comment:
the console was complaining about something already existing when `setButton` ran a second time. This `destroy` made it go away.